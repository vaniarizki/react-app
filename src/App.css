* {
  box-sizing: border-box;
}

body {
  font-family: Arial;
  padding: 10px;
  background: #eae4ef;
}

/* Header/Blog Title */
.header {
  padding: 15px;
  text-align: center;
  background: #a9b4d6;
}

.header h1 {
  font-size: 50px;
  color: #f4f1f7;
}

/* Style the top navigation bar */
.topnav {
  overflow: hidden;
  background-color: #9d8dac;
}

/* Style the topnav links */
.topnav a {
  float: left;
  display: block;
  color: #f2f2f2;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

/* Change color on hover */
.topnav a:hover{
  background-color: #dab0a5;
  color: whitesmoke;
}

/* Create two unequal columns that floats next to each other */
/* Left column */
.leftcolumn {   
  float: left;
  width: 100%;
  color: #3d3942;
  margin-bottom: 40px;
}

/* Add a card effect for articles */
.card {
  background-color: #dfd6e7;
  padding: 20px 20px 30px 20px;
  margin-top: 20px;
}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

/* Footer */
.footer {
  padding: 10px;
  text-align: center;
  background: #a9b4d6;
  margin-top: 20px;
  color: #2a1642;
  bottom: 0;
  position: fixed;
  right: 0;
  left: 0;
  overflow-x: hidden;
}
.footer a {
  color: #18161b;
  text-decoration: none;
}
/* Responsive layout - when the screen is less than 800px wide, make the two columns stack on top of each other instead of next to each other */
@media screen and (max-width: 800px) {
  .leftcolumn, .rightcolumn {   
    width: 100%;
    padding: 0;
  }
}

/* Responsive layout - when the screen is less than 400px wide, make the navigation links stack on top of each other instead of next to each other */
@media screen and (max-width: 400px) {
  .topnav a {
    float: none;
    width: 100%;
  }
}
/* Center */
.center {
  text-align: center;
}
.darker {
  color: #3d3942;
}
.input {
  width: 100%;
  padding: 7px 15px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
}
.submit {
  width: 100%;
  background-color: #696272;
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}
button[type=submit]:hover {
  background-color: #3d3942;
}
.form {
  max-width: 600px;
  width:50%; 
  margin-left:25%; 
  margin-right:25%;
}
.daftar {
  overflow-x: auto;
  margin-bottom: 50px;
}
.daftar table {
  width: 100%;
}
th {
  text-align: left;
}
td {
  padding: 5px 0;
}
.daftar tr:hover {background-color:#bfb0ce;}
td a {
  color: #3d3942;
  text-decoration: none;
}
.daftar th, .daftar td {
  padding: 8px;
  text-align: left;
  border-bottom: 1px solid #3d3942;
}
.icon {
  height: 21px;
  width: 21px;
}
table {border-collapse: collapse;}
.daftar button {
  border-style: none;
  cursor: pointer;
  background-color: transparent;
}
