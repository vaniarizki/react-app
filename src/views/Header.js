import React from 'react';

class Header extends React.Component {
  render() {
    return (
      <div className="header">
        <h1>Aplikasi Kontak</h1>
        <p className="darker" >Aplikasi yang berisi kontak yang tersambung dengan sosial media</p>
      </div>
    )
  }
}

export default Header;