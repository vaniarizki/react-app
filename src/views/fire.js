import firebase from 'firebase';
const firebaseConfig = {
  apiKey: "AIzaSyBldXW0IvoFqukXvlnnsyXJNeDNfuYsl0I",
  authDomain: "myfinalproject22.firebaseapp.com",
  databaseURL: "https://myfinalproject22.firebaseio.com",
  projectId: "myfinalproject22",
  storageBucket: "myfinalproject22.appspot.com",
  messagingSenderId: "113820924053",
  appId: "1:113820924053:web:4c22e8aea919af25c0439c"
};

var fire = firebase.initializeApp(firebaseConfig);
export default fire;