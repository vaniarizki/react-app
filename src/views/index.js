export { default as Header } from './Header';
export { default as Add } from './Add';
export { default as Contacts } from './Contacts';
export { default as Footer } from './Footer';
