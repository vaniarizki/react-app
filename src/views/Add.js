import React, { Component } from 'react';
import fire from './fire';
export let contacts = [];
export default class Add extends Component {
  constructor(props) {
    super(props);
    this.state = { contacts };
  }
  iName = React.createRef();
  iTel = React.createRef();
  iEmail = React.createRef();
  iWeb = React.createRef();
  iBday = React.createRef();
  iTwitter = React.createRef();
  iIg = React.createRef();
  onSubmit = (event) => {
    event.preventDefault();
    event.stopPropagation();
    let newContact = {
      name: `${this.iName.current.value}`,
      tel: `${this.iTel.current.value}`,
      email: `${this.iEmail.current.value}`,
      web: `${this.iWeb.current.value}`,
      bday: `${this.iBday.current.value}`,
      twitter: `${this.iTwitter.current.value}`,
      ig: `${this.iIg.current.value}`
    };
    fire.database().ref().child('contacts').push(newContact);
    alert('Kontak tersimpan');
    event.preventDefault();
    window.location.reload();
  }
  render() {
    return (
      <div className="leftcolumn card">
        <h2 className="center">Tambah Kontak Baru</h2>
        <form onSubmit={this.onSubmit}>
          <table className="form">
            <tbody>
              <tr>
                <td>Nama</td>
                <td>
                  <input className="input" type="text" name="name" ref={this.iName} required />
                </td>
              </tr>
              <tr>
                <td>Nomor Telp</td>
                <td>
                  <input className="input" type="tel" name="tel" ref={this.iTel} placeholder="081288888888" required />
                </td>
              </tr>
              <tr>
                <td>Email</td>
                <td>
                  <input className="input" type="email" name="email" ref={this.iEmail} />
                </td>
              </tr>
              <tr>
                <td>Website</td>
                <td>
                  <input className="input" type="text" name="web" ref={this.iWeb} placeholder="contoh: facebook.com" />
                </td>
              </tr>
              <tr>
                <td>Tanggal Lahir</td>
                <td>
                  <input className="input" type="date" name="bday" ref={this.iBday} />
                </td>
              </tr>
              <tr>
                <td>Twitter</td>
                <td>
                  <input className="input" type="text" name="twitter" ref={this.iTwitter} placeholder="tanpa '@'" />
                </td>
              </tr>
              <tr>
                <td>Instagram</td>
                <td>
                  <input className="input" type="text" name="ig" ref={this.iIg} placeholder="tanpa '@'" />
                </td>
              </tr>
              <tr>
                <td className="center" colSpan="2">
                  <button className="submit" type="submit">Simpan</button>
                </td>
              </tr>
            </tbody>
          </table>
        </form>
      </div>
    )
  }
}