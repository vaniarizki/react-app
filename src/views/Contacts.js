import React, { useEffect, useState } from 'react';
import fire from './fire';
import { default as Wa } from '../icon/whatsapp.png';
import { default as Del } from '../icon/delete.png';

// menampilkan daftar kontak

const th = ['Nama', 'No Telp', 'Email', 'Website', 'Tanggal Lahir', 'Twitter', 'Instagram', 'WA', 'Aksi'];
// handling delete
const handleDelete = (id, name) => {
  let c = window.confirm(`Hapus kontak ${name}?`)
  if (c === true) {
    fire.database().ref().child(`contacts/${id}`).remove()
    alert(`Berhasil menghapus kontak ${name}`);
  } else {
    alert(`Kontak ${name} tidak dihapus`)
  }
}

export default function Contacts() {
  let [contactObj, setContactObj] = useState({});
  let [count, setCount] = useState(0);
  // mengambil data dari database saat memuat halaman
  useEffect(() => {
    fire.database().ref().child('contacts').on('value', snapshot => {
      if (snapshot.val() != null) {
        setContactObj({ ...snapshot.val() });
        const count = snapshot.numChildren();
        setCount(count);
      }
    })
  }, [])

  return (
    <>
      <div className="leftcolumn card" >
        <h2 className="center">Daftar Kontak</h2>
        <div className="daftar">
          <p>Jumlah kontak: {count}</p>
          <table>
            <thead>
              <tr>{th.map(e => <th key={e}>{e}</th>)}</tr>
            </thead>
            <tbody>
              {Object.keys(contactObj).map((id) => (
                <tr key={id}>
                  <td>{contactObj[id].name}</td>
                  <td>{contactObj[id].tel}</td>
                  <td><a href={`mailto:${contactObj[id].email}`} title={`Kirim email ke ${contactObj[id].name}`}>{contactObj[id].email}</a></td>
                  <td>{contactObj[id].web}</td>
                  <td>{contactObj[id].bday}</td>
                  <td><a target="_blank" rel="noopener noreferrer" href={`https://twitter.com/${contactObj[id].twitter}`}>
                    {contactObj[id].twitter}
                  </a></td>
                  <td><a target="_blank" rel="noopener noreferrer" href={`https://www.instagram.com/${contactObj[id].ig}`}>
                    {contactObj[id].ig}
                  </a></td>
                  <td><a target="_blank" rel="noopener noreferrer" href={`https://wa.me/62${contactObj[id].tel}`}>
                    <img className="icon" alt='WhatsApp' title={`Klik untuk membuka WhatsApp ${contactObj[id].name}`} src={Wa}></img>
                  </a></td>
                  <td>
                    <button onClick={() => handleDelete(id, `${contactObj[id].name}`)}>
                      <img className="icon" alt='Hapus' title={`Menghapus kontak ${contactObj[id].name}`} src={Del}></img>
                    </button>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div >
      </div >
    </>
  )
}


