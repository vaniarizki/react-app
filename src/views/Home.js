import React, { useEffect, useState } from 'react';
import fire from './fire';
export default function Home() {
  let [count, setCount] = useState(0);
  useEffect(() => {
    fire.database().ref().child('contacts').on('value', snapshot => {
      if (snapshot.val() != null) {
        const count = snapshot.numChildren();
        setCount(count)
      }
    })
  }, [])
  return (
    <>

    </>
  )
}