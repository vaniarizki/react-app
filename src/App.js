import React from 'react';
import './App.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from 'react-router-dom';
import { Header, Add, Contacts, Footer } from './views';

function App() {
  return (
    <>
      {/* Menampilkan Header */}
      <Header />
      {/* Router pages */}
      <Router>
        <div className="topnav">
          <Link to="/">Daftar Kontak</Link>
          <Link to="/add">Tambah Kontak</Link>
        </div>
        <div>
          <Switch>
            <Route exact path="/">
              <Contacts />
            </Route>
            <Route exact path="/add">
              <Add />
            </Route>
          </Switch>
        </div>
      </Router >
      <Footer />
    </>
  );
}

export default App;
