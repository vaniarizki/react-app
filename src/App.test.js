import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import { Header, Add, Contacts, Footer } from './views';
import fire from './views/fire';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Jumlah kontak/i);
  expect(linkElement).toBeInTheDocument();
});

test('test1', () => {
  const component = renderer.create(
    <App />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
test('test2', () => {
  const component = renderer.create(
    <Header />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
test('test3', () => {
  const component = renderer.create(
    <Add />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.getInstance().onSubmit;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
test('test4', () => {
  const component = renderer.create(
    <Contacts />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
test('test5', () => {
  const component = renderer.create(
    <Footer />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
test('test6', () => {
  const component = renderer.create(
    <fire />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})