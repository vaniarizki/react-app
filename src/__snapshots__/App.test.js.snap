// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test1 1`] = `
Array [
  <div
    className="header"
  >
    <h1>
      Aplikasi Kontak
    </h1>
    <p
      className="darker"
    >
      Aplikasi yang berisi kontak yang tersambung dengan sosial media
    </p>
  </div>,
  <div
    className="topnav"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Daftar Kontak
    </a>
    <a
      href="/add"
      onClick={[Function]}
    >
      Tambah Kontak
    </a>
  </div>,
  <div>
    <div
      className="leftcolumn card"
    >
      <h2
        className="center"
      >
        Daftar Kontak
      </h2>
      <div
        className="daftar"
      >
        <p>
          Jumlah kontak: 
          0
        </p>
        <table>
          <thead>
            <tr>
              <th>
                Nama
              </th>
              <th>
                No Telp
              </th>
              <th>
                Email
              </th>
              <th>
                Website
              </th>
              <th>
                Tanggal Lahir
              </th>
              <th>
                Twitter
              </th>
              <th>
                Instagram
              </th>
              <th>
                WA
              </th>
              <th>
                Aksi
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>,
  <div
    className="footer"
  >
    © Copyright 2020. Icons made by 
    <a
      href="https://www.flaticon.com/authors/freepik"
      title="Freepik"
    >
      Freepik
    </a>
     from 
    <a
      href="https://www.flaticon.com/"
      title="Flaticon"
    >
      www.flaticon.com
    </a>
  </div>,
]
`;

exports[`test2 1`] = `
<div
  className="header"
>
  <h1>
    Aplikasi Kontak
  </h1>
  <p
    className="darker"
  >
    Aplikasi yang berisi kontak yang tersambung dengan sosial media
  </p>
</div>
`;

exports[`test3 1`] = `
<div
  className="leftcolumn card"
>
  <h2
    className="center"
  >
    Tambah Kontak Baru
  </h2>
  <form
    onSubmit={[Function]}
  >
    <table
      className="form"
    >
      <tbody>
        <tr>
          <td>
            Nama
          </td>
          <td>
            <input
              className="input"
              name="name"
              required={true}
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td>
            Nomor Telp
          </td>
          <td>
            <input
              className="input"
              name="tel"
              placeholder="081288888888"
              required={true}
              type="tel"
            />
          </td>
        </tr>
        <tr>
          <td>
            Email
          </td>
          <td>
            <input
              className="input"
              name="email"
              type="email"
            />
          </td>
        </tr>
        <tr>
          <td>
            Website
          </td>
          <td>
            <input
              className="input"
              name="web"
              placeholder="contoh: facebook.com"
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td>
            Tanggal Lahir
          </td>
          <td>
            <input
              className="input"
              name="bday"
              type="date"
            />
          </td>
        </tr>
        <tr>
          <td>
            Twitter
          </td>
          <td>
            <input
              className="input"
              name="twitter"
              placeholder="tanpa '@'"
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td>
            Instagram
          </td>
          <td>
            <input
              className="input"
              name="ig"
              placeholder="tanpa '@'"
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td
            className="center"
            colSpan="2"
          >
            <button
              className="submit"
              type="submit"
            >
              Simpan
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;

exports[`test3 2`] = `
<div
  className="leftcolumn card"
>
  <h2
    className="center"
  >
    Tambah Kontak Baru
  </h2>
  <form
    onSubmit={[Function]}
  >
    <table
      className="form"
    >
      <tbody>
        <tr>
          <td>
            Nama
          </td>
          <td>
            <input
              className="input"
              name="name"
              required={true}
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td>
            Nomor Telp
          </td>
          <td>
            <input
              className="input"
              name="tel"
              placeholder="081288888888"
              required={true}
              type="tel"
            />
          </td>
        </tr>
        <tr>
          <td>
            Email
          </td>
          <td>
            <input
              className="input"
              name="email"
              type="email"
            />
          </td>
        </tr>
        <tr>
          <td>
            Website
          </td>
          <td>
            <input
              className="input"
              name="web"
              placeholder="contoh: facebook.com"
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td>
            Tanggal Lahir
          </td>
          <td>
            <input
              className="input"
              name="bday"
              type="date"
            />
          </td>
        </tr>
        <tr>
          <td>
            Twitter
          </td>
          <td>
            <input
              className="input"
              name="twitter"
              placeholder="tanpa '@'"
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td>
            Instagram
          </td>
          <td>
            <input
              className="input"
              name="ig"
              placeholder="tanpa '@'"
              type="text"
            />
          </td>
        </tr>
        <tr>
          <td
            className="center"
            colSpan="2"
          >
            <button
              className="submit"
              type="submit"
            >
              Simpan
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;

exports[`test4 1`] = `
<div
  className="leftcolumn card"
>
  <h2
    className="center"
  >
    Daftar Kontak
  </h2>
  <div
    className="daftar"
  >
    <p>
      Jumlah kontak: 
      0
    </p>
    <table>
      <thead>
        <tr>
          <th>
            Nama
          </th>
          <th>
            No Telp
          </th>
          <th>
            Email
          </th>
          <th>
            Website
          </th>
          <th>
            Tanggal Lahir
          </th>
          <th>
            Twitter
          </th>
          <th>
            Instagram
          </th>
          <th>
            WA
          </th>
          <th>
            Aksi
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`test5 1`] = `
<div
  className="footer"
>
  © Copyright 2020. Icons made by 
  <a
    href="https://www.flaticon.com/authors/freepik"
    title="Freepik"
  >
    Freepik
  </a>
   from 
  <a
    href="https://www.flaticon.com/"
    title="Flaticon"
  >
    www.flaticon.com
  </a>
</div>
`;

exports[`test6 1`] = `<fire />`;
